server/service/grpc/service.go

```diff
@@ -48,6 +48,10 @@ type Handler interface {
 
 // NodeHeartbeat implements gRPC CeresmetaServer.
 func (s *Service) NodeHeartbeat(ctx context.Context, req *metaservicepb.NodeHeartbeatRequest) (*metaservicepb.NodeHeartbeatResponse, error) {
+	if ok, err := s.allow(ctx); !ok {
+		return nil, err
```

Comment:
```suggestion
		return &metaservicepb.NodeHeartbeatResponse{Header: responseHeader(err, "grpc heartbeat")}, nil
```
return `NodeHeartbeatResponse` rather than origin error.